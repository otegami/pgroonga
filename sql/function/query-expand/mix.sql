CREATE TABLE synonyms (
  term text PRIMARY KEY,
  synonyms text[]
);

INSERT INTO synonyms VALUES ('Mroonga', ARRAY['Mroonga', 'Groonga MySQL']);
INSERT INTO synonyms VALUES ('PGroonga', ARRAY['PGroonga', 'Groonga PostgreSQL']);

SELECT pgroonga_query_expand('synonyms', 'term', 'synonyms',
                             'Ruby (Mroonga OR PGroonga)');

DROP TABLE synonyms;
