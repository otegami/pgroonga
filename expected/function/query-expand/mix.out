CREATE TABLE synonyms (
  term text PRIMARY KEY,
  synonyms text[]
);
INSERT INTO synonyms VALUES ('Mroonga', ARRAY['Mroonga', 'Groonga MySQL']);
INSERT INTO synonyms VALUES ('PGroonga', ARRAY['PGroonga', 'Groonga PostgreSQL']);
SELECT pgroonga_query_expand('synonyms', 'term', 'synonyms',
                             'Ruby (Mroonga OR PGroonga)');
                             pgroonga_query_expand                             
-------------------------------------------------------------------------------
 Ruby (((Mroonga) OR (Groonga MySQL)) OR ((PGroonga) OR (Groonga PostgreSQL)))
(1 row)

DROP TABLE synonyms;
